cinder/volume/manager.py

```diff
@@ -1345,6 +1345,18 @@ def create_snapshot(self, context, snapshot) -> ovo_fields.UUIDField:
         snapshot.encryption_key_id = vol_ref.encryption_key_id
         snapshot.save()
 
+        # For snapshots that have a hidden backend key
+        # We update the allocated capacity to account
+        # for the snapshot size on the backend.
+        # the hidden backend key is there for snapshots
+        # that are clones.
+        key = objects.snapshot.SAP_HIDDEN_BACKEND_KEY
+        if snapshot.metadata:
```

Comment:
Does a snapshot ever _not_ have metadata? From `cinder/objects/snapshot.py:85-86`:

```python
    def _get_expected_attrs(cls, context, *args, **kwargs):
        return 'metadata',
```

So it should always have it?

Reply:
yes sometimes there is no metadata.